SmartDeviceLinkTests/TransportSpecs/SDLTCPTransportSpec.m

```diff
@@ -0,0 +1,656 @@
+//
+//  SDLTCPTransportSpec.m
+//  SmartDeviceLinkTests
+//
+//  Created by Sho Amano on 2018/04/24.
+//  Copyright © 2018 Xevo Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <Quick/Quick.h>
+#import <Nimble/Nimble.h>
+#import <OCMock/OCMock.h>
+
+#import "SDLTCPTransport.h"
+#import "SDLError.h"
+
+#import <sys/types.h>
+#import <sys/socket.h>
+#import <netdb.h>
+#import <sys/select.h>
+#import <sys/time.h>
+#import <fcntl.h>
+#import <stdio.h>
+#import <string.h>
+#import <errno.h>
+
+#define RECV_BUF_SIZE           (1024)
+#define MAX_SERVER_SOCKET_NUM   (16)
+#define THREAD_STOP_WAIT_SEC    (1.0)
+
+@protocol TestTCPServerDelegate
+- (void)onClientConnected;
+- (void)onClientDataReceived:(NSData *)data;
+- (void)onClientShutdown;
+- (void)onClientError;
+@end
+
+@interface TestTCPServer : NSObject {
```

Comment:
Please put this class in its own file(s), similar to how the `TestConnectionManager` is

Reply:
OK, done in commit 713f409d85e145e1bd3caa495371477f61edd855.